blockchain/blockchain.go

```diff
@@ -901,7 +899,7 @@ func (bc *blockchain) commitBlock(blk *block.Block) error {
 	}
 	// write block into DB
 	putTimer := bc.timerFactory.NewTimer("putBlock")
-	if err = bc.dao.PutBlock(blk); err == nil {
+	if err = bc.dao.PutBlock(blk); err != nil {
```

Comment:
in case PutBlock() does return an err != nil, err will be overwritten since dao.Commit() always return nil
so should use == here

actually would suggest we remove the Commit() interface from DAO for 2 reasons:

1. I know the intention is that DAO can be also considered an Indexer so it also have Commit() to unify interface, but DAO is a very basic module to put raw blocks, it won't be actually used as Indexer

2. Indexer's DeleteTipBlock(*Block) needs an input block, while DAO's DeleteTipBlock() does not, so anyway DAO has a different interface from Indexer now